File: docs/sphinx/snippets/python/default_ops/rz_op.py
import cudaq
import math

def main():
    qubit = cudaq.qubit()
    # [Begin Rz Op]
    # Apply the unitary transformation
    # Rz(λ) = | exp(-iλ/2)      0     |
    #         |     0       exp(iλ/2) |
    rz(math.pi, qubit)
    # [End Rz Op]
if __name__ == "__main__":
    main()

File: docs/sphinx/snippets/cpp/default_ops/rz_op.cpp
/*******************************************************************************
 * Copyright (c) 2022 - 2025 NVIDIA Corporation & Affiliates.                  *
 * All rights reserved.                                                        *
 *                                                                             *
 * This source code and the accompanying materials are made available under    *
 * the terms of the Apache License 2.0 which accompanies this distribution.    *
 ******************************************************************************/

// Compile and run with: `nvq++ rz_op.cpp && ./a.out`

#include <cudaq.h>

int main() {
  cudaq::qubit qubit;

  // Apply the unitary transformation
  // Rz(λ) = | exp(-iλ/2)      0     |
  //         |     0       exp(iλ/2) |
  rz(std::numbers::pi, qubit);

  return 0;
}

